src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs

```diff
@@ -624,13 +624,13 @@ public async Task ReadISetTOfHashSetT()
 
             if (result.First().Contains(1))
             {
-                Assert.Equal(new HashSet<int> { 1, 2 }, result.First());
-                Assert.Equal(new HashSet<int> { 3, 4 }, result.Last());
+                Assert.True(result.First().SetEquals(new []{ 1, 2 }));
```

Comment:
If you paste the stack I can point you to the RD.XML to fix this. I've seen this, it's fixable.

The problem will also go away once we ingest AOT compatible xunit asserts (https://github.com/dotnet/arcade/pull/13241).

Maybe it would be preferable to fix this with RD.XML to keep good diagnostics for failures.